lib/lib-openapi/src/test/java/org/sagebionetworks/translator/ControllerModelDocletTest.java

```diff
@@ -0,0 +1,222 @@
+		JSONObject pathObj = pathsObj.getJSONObject("repo/v1/complex-pet/{petName}");
+		JSONObject operationObj = pathObj.getJSONObject("get");
+		
+		// test parameters are correct
+		JSONArray parameters = operationObj.getJSONArray("parameters");
+		boolean foundPetNameParam = false;
+		for (int i = 0; i < parameters.length(); i++) {
+			JSONObject param = parameters.getJSONObject(i);
+			if (param.getString("name").equals("petName")) {
+				assertEquals("path", param.getString("in"));
+				assertEquals(true, param.getBoolean("required"));
+				JSONObject schema = param.getJSONObject("schema");
+				assertEquals("string", schema.getString("type"));
+				foundPetNameParam = true;
+			}
+		}
+		assertTrue(foundPetNameParam);
+	}
+	
+	@Test
+	public void testResponseObjectIsCorrectInPath() {
+		JSONObject pathsObj = generatedOpenAPISpec.getJSONObject("paths");
+		JSONObject pathObj = pathsObj.getJSONObject("repo/v1/complex-pet/{petName}");
+		JSONObject operationObj = pathObj.getJSONObject("get");
+		
+		// test response object is correct
+		JSONObject response = operationObj.getJSONObject("responses");
+		JSONObject responseStatusCode = response.getJSONObject("200");
+		assertEquals("the Pet associated with 'name'.", responseStatusCode.getString("description"));
+		JSONObject content = responseStatusCode.getJSONObject("content");
+		JSONObject contentType = content.getJSONObject("application/json");
+		System.out.println("content type JSONObject: " + contentType.toString(5));
+		JSONObject responseSchema = contentType.getJSONObject("schema");
+
+		// test to see if Pet interface is represented correctly as the return type.
+		assertEquals("object", responseSchema.getString("type"));
+		assertEquals("This interface represents a pet.", responseSchema.getString("description"));
+		JSONObject responseSchemaProperties = responseSchema.getJSONObject("properties");
+		JSONObject responseSchemaNameProperty = responseSchemaProperties.getJSONObject("name");
+		assertEquals("string", responseSchemaNameProperty.getString("type"));
+		JSONObject responseSchemaHasTailProperty = responseSchemaProperties.getJSONObject("hasTail");
+		assertEquals("_boolean", responseSchemaHasTailProperty.getString("type"));
+		
+		// test to see if the oneof property is being set correctly.
+		JSONArray oneOf = responseSchema.getJSONArray("oneOf");
+		assertTrue(oneOf.length() == 3);
+		boolean foundHusky = false;
+		boolean foundPoodle = false;
+		boolean foundCat = false;
+		for (int i = 0; i < oneOf.length(); i++) {
+			JSONObject reference = oneOf.getJSONObject(i);
+			String ref = reference.getString("$ref");
+			if (ref.equals("#/components/org.sagebionetworks.openapi.pet.Husky")) {
+				foundHusky = true;
+			} else if (ref.equals("#/components/org.sagebionetworks.openapi.pet.Poodle")) {
+				foundPoodle = true;
+			} else if (ref.equals("#/components/org.sagebionetworks.openapi.pet.Cat")) {
+				foundCat = true;
+			}
```

Comment:
if you add each string to a set, you can check the count and do set.contains("should be there") for each value.